src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1017,10 +1033,14 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
                 {
                     // At this point we have a compilation with nothing yet computed. 
                     // We pass it to the generators, which will realize any symbols they require. 
-                    compilation = RunGenerators(compilation, Arguments.ParseOptions, generators, analyzerConfigProvider, additionalTextFiles, diagnostics);
+                    (compilation, var timingInfo) = RunGenerators(compilation, Arguments.ParseOptions, generators, analyzerConfigProvider, additionalTextFiles, Arguments.ReportAnalyzer, diagnostics);
```

Comment:
⚠️ I do not believe we should alter the runtime behavior in response to `ReportAnalyzer`. Otherwise, we are required to let users known (in some cases) that the generators are slow _because_ they tried to measure them. 

One solution to slow generators is to enable the cache, but with this behavior it is impossible to validate.

Reply:
@chsienki doing this would require that `GeneratorDriverState.With` accept `TrackIncrementalSteps` as one of the values that can be overridden. Today it does not and have to assume there is a reason for that. Can it be safely changed in a cached driver or not?

Reply:
I'm not sure I understand why you need to change it. From the looks of the PR it appears you're turning on `TrackIncrementalGeneratorSteps` in order to use the timing API but the two things are orthogonal. 

The timing API is always available, and you will always get a result from `driver.GetTimingInfo()`. 

`TrackIncrementalGeneratorSteps` turns on the granular step tracking which allows a user to inspect the individual step timing as well as inputs and outputs, which populates extra fields inside the struct returned from `driver.GetRunResult()`. That is a perf hit, so it's always disabled in the compiler, and instead users can into it in unit tests if they want to assert things about the pipeline.